hbase-server/src/main/java/org/apache/hadoop/hbase/conf/ConfigurationManager.java

```diff
@@ -137,4 +138,14 @@ public int getNumObservers() {
       return configurationObservers.size();
     }
   }
+
+  /**
+   * @return the observers
+   */
+  @VisibleForTesting
+  public Set<ConfigurationObserver> getConfigurationObservers() {
```

Comment:
This is used by tests only.  But still how this synchronized block helps here?  We are directly returning that object which is not thread safe. IMO we should add a method like containObserver (or a better name?) and just return a boolean.

Reply:
Good point! The synchronization here doesn't prevent multithread-access afterwards, right? Or we may not consider the multi-thread access cos it is used in unittest only? What do you think about it, Anoops ? I added a synchronized method which returns a boolean for now, looking forward to your opinion.

Reply:
Here the boolean based API is enough.  That is what u did now.
Nit change method name   containObserver  -> containsObserver

Reply:
Yeah, thanks